rules/mapping.smk

```diff
@@ -48,6 +48,22 @@ rule recalibrate_base_qualities:
     wrapper:
         get_wrapper_path("gatk", "baserecalibrator")
 
+rule remove_decoy:
+    input:
+        bam = "recal/{sample}-{unit}.bam",
+        decoy = config["remove_decoy"]["bed"] 
+    output: 
+        out_rm = "decoy_rm/{sample}-{unit}.removed_reads.bam",
+        out_f = "decoy_rm/{sample}-{unit}.no_decoy_reads.bam"
+    log:
+        "logs/remove_decoys/{sample}-{unit}.log"
+    threads: 8
+    resources: mem_mb=10000
+    wrapper:
+        get_wrapper_path("samtools", "remove_decoys") # or whatever 
```

Comment:
Wrappers is meant to wrap around a single tool. In this case you've created/ added some new functionality to samtools view. I'm not so sure if its the most appropriate use of a wrapper script. 

I left a comment on the wrapper, noting that we could use samtools view and samtools reheader to achieve the read removal step. How about creating wrappers for both these tools and writing two rules?

What do you think?

Reply:
Yeah I've been thinking about this and you're right, it is a bit more niche and doesn't fit the use case for a wrapper so I'd say just a shell command in the rule for now. Definitely open to more discussion though if you think otherwise.